.github/workflows/main.yml

```diff
@@ -0,0 +1,39 @@
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
```

Comment:
Should we let this run on all PRs and branches?

Reply:
I think that this covers the vast majority of cases but yeah, we can let it run everywhere.